pkg/kwokctl/cmd/create/cluster/cluster.go

```diff
@@ -127,15 +127,42 @@ func runE(ctx context.Context, flags *flagpole) error {
 		defer cancel()
 	}
 
-	buildRuntime, ok := runtime.DefaultRegistry.Get(flags.Options.Runtime)
-	if !ok {
-		return fmt.Errorf("runtime %q not found", flags.Options.Runtime)
-	}
-
-	rt, err := buildRuntime(name, workdir)
-	if err != nil {
-		return err
+	var rt runtime.Runtime
+	var err error
+	if flags.Options.Runtime == "" {
+		errs := make([]error, 0, len(flags.Options.Runtimes))
+		for _, r := range flags.Options.Runtimes {
+			buildRuntime, ok := runtime.DefaultRegistry.Get(r)
+			if !ok {
+				err = fmt.Errorf("runtime %q not found", flags.Options.Runtime)
+				errs = append(errs, err)
+				continue
+			}
+			if rt, err = buildRuntime(name, workdir); err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			if err = rt.Available(ctx); err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			flags.Options.Runtime = r
+			break
+		}
+		if flags.Options.Runtime == "" {
+			return fmt.Errorf("runtime %v not available: %v", flags.Options.Runtimes, errs)
+		}
+	} else {
+		buildRuntime, ok := runtime.DefaultRegistry.Get(flags.Options.Runtime)
+		if !ok {
+			return fmt.Errorf("runtime %q not found", flags.Options.Runtime)
+		}
+		rt, err = buildRuntime(name, workdir)
+		if err != nil {
+			return fmt.Errorf("runtime %v not available: %w", flags.Options.Runtime, err)
+		}
 	}
+	logger = logger.With("runtime", flags.Options.Runtime)
```

Comment:
```suggestion
```

Reply:
as we now detect runtime automatically, it maybe worth logging the runtime?